### PR TITLE
ccl 2021 ingestion

### LINKED ISSUE #1532: Ingest CCL 2021
https://www.dropbox.com/s/qji3pa7mt3p1vv3/ccl2021_data.tgz?dl=0

### PR DESCRIPTION
**Pull request by xinru1414**
ingested ccl 2021, closes #1532

### CODE REVIEW

**Comment by xinru1414** [comment]
@mjpost not sure which sig file to add this to?

**Comment by mjpost** [comment]
The builds seem to fail on the `docopt` package lately. Can one of you [take a look at this](https://github.com/acl-org/acl-anthology/pull/1533/checks?check_run_id=3340490610)? (Maybe @akoehn can help?)

```
Building wheels for collected packages: docopt, iso-639, stop-words, texsoup, coverage
  Building wheel for docopt (setup.py): started
  Building wheel for docopt (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /home/runner/work/acl-anthology/acl-anthology/venv/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0jzwd4ia/docopt/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0jzwd4ia/docopt/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-l2pecl33
       cwd: /tmp/pip-install-0jzwd4ia/docopt/
  Running setup.py clean for docopt
```

**Comment by mbollmann** [comment]
I'd say this is the reason for the failing build...

```
ERROR    Paper 2021.ccl-1.0 uses ambiguous name 'Yang Liu' without id
ERROR      Please add an id, for example: yang-liu-edinburgh yang-liu-icsi yang-liu-ict yang-liu yang-liu-umich
```

**Comment by mjpost** [comment]
Ah thank you! @xinru1414 I bet the ingestion code doesn't resolve ambiguities on `<editor>`, can you fix that?

File: data/xml/2021.ccl.xml [hunk 1]
```diff
@@ -0,0 +1,1637 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<collection id="2021.ccl">
+  <volume id="1" ingest-date="2021-08-31">
+    <meta>
+      <booktitle>"Proceedings of the 20th Chinese National Conference on Computational Linguistics"</booktitle>
```
Comment (mjpost) [inline data/xml/2021.ccl.xml:5]: There shouldn’t be quotes here.

File: data/xml/2021.ccl.xml [hunk 2]
```diff
@@ -0,0 +1,1637 @@
...
+      <editor><first>Gaoqi</first><last>Rao</last></editor>
+      <editor><first>生</first><last>李</last></editor>
+      <editor><first>茂松</first><last>孙</last></editor>
+      <editor><first>洋</first><last>刘</last></editor>
+      <editor><first>华</first><last>吴</last></editor>
+      <editor><first>康</first><last>刘</last></editor>
+      <editor><first>万翔</first><last>车</last></editor>
+      <editor><first>世柱</first><last>何</last></editor>
+      <editor><first>高琦</first><last>饶</last></editor>
+      <publisher>Chinese Information Processing Society of China</publisher>
```
Comment (mjpost) [inline data/xml/2021.ccl.xml:22]: None of this looks right. It looks like the Chinese names have been added as extra authors. They should be added as `<variant>` tags. See CCL 2020 for how it’s done.

File: data/xml/2021.ccl.xml (hunk 1 above)
Comment (xinru1414) [reply data/xml/2021.ccl.xml:5]: Yep got it

File: data/xml/2021.ccl.xml (hunk 2 above)
Comment (xinru1414) [reply data/xml/2021.ccl.xml:22]: Okay, I'll look into this

**Comment by xinru1414** [comment]
> Ah thank you! @xinru1414 I bet the ingestion code doesn't resolve ambiguities on `<editor>`, can you fix that?

@mjpost It looks like your simple fix `resolved name ambiguity 1f8daea` resolved the issue (all checks are passing)?

**Comment by mjpost** [comment]
Yes, unresolved ambiguous names (authors or editors) will break the build.

Do you know what the requested ingestion date is for this?

**Comment by mjpost** [comment]
[Here is the thread](https://github.com/acl-org/acl-anthology/pull/1027) where we did this last year. Unfortunately I don’t remember how I did this, if it was manual or if I adjusted the scripts. You might be able to find notes or code in the CCL 2020 ingestion folder on Dropbox.

**Comment by xinru1414** [comment]
> Yes, unresolved ambiguous names (authors or editors) will break the build.
> 
> Do you know what the requested ingestion date is for this?

Yes, Aug 31st

**Comment by xinru1414** [comment]
> [Here is the thread](https://github.com/acl-org/acl-anthology/pull/1027) where we did this last year. Unfortunately I don’t remember how I did this, if it was manual or if I adjusted the scripts. You might be able to find notes or code in the CCL 2020 ingestion folder on Dropbox.

Ok sounds good I'll look into this

**Comment by xinru1414** [comment]
@mjpost the quick-fix work flow for CCL ingestion right now is 

- run ingest.py
- run author_names_to_variants.py
- run write_bibkeys_to_xml.py

Are there any other venues that need `author_names_to_variants.py` that you can think of?

**Comment by xinru1414** [comment]
> The builds seem to fail on the `docopt` package lately. Can one of you [take a look at this](https://github.com/acl-org/acl-anthology/pull/1533/checks?check_run_id=3340490610)? (Maybe @akoehn can help?)
> 
> ```
> Building wheels for collected packages: docopt, iso-639, stop-words, texsoup, coverage
>   Building wheel for docopt (setup.py): started
>   Building wheel for docopt (setup.py): finished with status 'error'
>   ERROR: Command errored out with exit status 1:
>    command: /home/runner/work/acl-anthology/acl-anthology/venv/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0jzwd4ia/docopt/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0jzwd4ia/docopt/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-l2pecl33
>        cwd: /tmp/pip-install-0jzwd4ia/docopt/
>   Running setup.py clean for docopt
> ```

@mjpost It seems like whenever the build fails it always throws this error, however it doesn't seem to give the real error? Or maybe I'm missing something? 

For example in this thread my earlier commits (`348f940` and `fc0dd2c`) caused this error because of the name variant. My later commits (`be430c1` and `8597d1d`) caused the same error because ingest.py wasn't reformatted (which I didn't know what was wrong until I looked at the commit and figured out that there is an extra line in the code)

**Comment by mbollmann** [comment]
> @mjpost It seems like whenever the build fails it always throws this error, however it doesn't seem to give the real error? Or maybe I'm missing something?

It also throws these errors when the build succeeds. They can be safely ignored and are not the cause for the build failing. To see why a build fails, I always check the build log from the bottom up, as usually whatever is causing it appears at the end of the log.

**Comment by xinru1414** [comment]
@mbollmann okay now I see the real error. Thanks for the tip!

File: bin/author_names_to_variants.py [hunk 3]
```diff
@@ -1,3 +1,9 @@
+'''
```
[the remark this answers is not in the document]
Comment (xinru1414) [reply bin/author_names_to_variants.py:1]: done

File: data/xml/2021.ccl.xml [hunk 4]
```diff
@@ -0,0 +1,1629 @@
...
+      <year>2021</year>
+    </meta>
+    <frontmatter>
+      <url hash="9c39efdc">2021.ccl-1.0</url>
+      <bibkey>ccl-2021-chinese</bibkey>
+    </frontmatter>
+    <paper id="1">
+      <title>融合零指代识别的篇章级机器翻译(Context-aware Machine Translation Integrating Zero Pronoun Recognition)</title>
+      <author><first>Hao</first><last>Wang</last><variant script="hani"><first>浩</first><last>汪</last></variant>
+      </author>
```
[the remark this answers is not in the document]
Comment (xinru1414) [reply data/xml/2021.ccl.xml:26]: Yes it did

**Review by mjpost — APPROVED** [review]

**Pull request merged**